modules/plugins/jboss-as-7/src/main/java/org/rhq/modules/plugins/jbossas7/JBossProductDiscoveryCallback.java

```diff
@@ -0,0 +1,159 @@
+/*
+ * RHQ Management Platform
+ * Copyright (C) 2005-2015 Red Hat, Inc.
+ * All rights reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation version 2 of the License.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software Foundation, Inc.,
+ * 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA
+ */
+package org.rhq.modules.plugins.jbossas7;
+
+import java.io.File;
+
+import org.rhq.core.pluginapi.inventory.DiscoveredResourceDetails;
+import org.rhq.core.pluginapi.inventory.ResourceDiscoveryCallback;
+import org.rhq.modules.plugins.jbossas7.helper.HostConfiguration;
+import org.rhq.modules.plugins.jbossas7.helper.HostPort;
+import org.rhq.modules.plugins.jbossas7.helper.ServerPluginConfiguration;
+
+/**
+ * Default abstract implementation of {@link ResourceDiscoveryCallback} interface, that may be extended by
+ * AS7 based plugins. This implementation handles cases when discovered JBossProduct equals to {@link JBossProduct#unknown()} - this means JBossAS7 plugin
+ * does know definition of JBossProduct it detected. The extending class only needs to implement {@link #getProduct()}.
+ * @author lzoubek@redhat.com
+ *
+ */
```

Comment:
On Thu, 28 May 2015 10:58:19 +0200, Thomas Segismont  
<notifications@github.com> wrote:

> Do you expect different usages of this class? If we only expect  
> implementors to override `getProduct` , then let's not expose the rest  
> of the class, and document here in the class Javadoc what's the class  
> behavior.
>
```getProduct``` is just the minimum. I also expect known product (like  
ISPN,JDV) plugins, could us this as base class, once they need (at least  
it gives them a bit more AS7 specific context). Thatswhy I did most of the  
stuff as protected. For example if gatein discovery callback is rewritten  
it can consume ```ServerPluginConfiguration.setExpectedRuntimeName```  
instead of (right now) hardcoded ```expectedRuntimeName``` string.
> ---
> Reply to this email directly or view it on GitHub:
> https://github.com/rhq-project/rhq/pull/172/files#r31214536


-- 
Libor Zoubek